services/device-registry-base/src/main/java/org/eclipse/hono/service/management/credentials/X509CertificateCredential.java

```diff
@@ -85,38 +82,46 @@ private X509CertificateCredential(final String distinguishedName, final List<X50
     @JsonCreator(mode = Mode.PROPERTIES)
     public static X509CertificateCredential fromProperties(
             @JsonProperty(value = RegistryManagementConstants.FIELD_PAYLOAD_CERT) final byte[] derEncodedX509Certificate,
-            @JsonProperty(value = RegistryManagementConstants.FIELD_AUTH_ID) final String distinguishedName,
+            @JsonProperty(value = RegistryManagementConstants.FIELD_AUTH_ID) final String authId,
             @JsonProperty(value = RegistryManagementConstants.FIELD_SECRETS) final List<X509CertificateSecret> secrets) {
 
         if (derEncodedX509Certificate == null) {
-            Objects.requireNonNull(distinguishedName);
+            Objects.requireNonNull(authId);
             Objects.requireNonNull(secrets);
             if (secrets.size() != 1) {
                 throw new IllegalArgumentException("list must contain exactly one secret");
             }
-            return fromSubjectDn(distinguishedName, secrets);
+            return fromAuthId(authId, secrets);
         } else {
             return fromCertificate(deserialize(derEncodedX509Certificate));
         }
     }
 
     /**
-     * Creates a new credentials object for an X.500 Distinguished Name.
+     * Creates a new credentials object for an authentication identifier.
      * <p>
-     * The given distinguished name will be normalized to RFC 2253 format.
+     * This method tries to normalize the authId in <a href="https://www.ietf.org/rfc/rfc2253.txt">RFC 2253</a> format.
+     * If that is not possible, the given authId is used.
      *
-     * @param distinguishedName The DN to use as the authentication identifier.
+     * @param authId The raw authentication identifier.
      * @param secrets The credential's secret(s).
      * @throws NullPointerException if any of the parameters are {@code null}.
-     * @throws IllegalArgumentException if the given string is not a valid X.500 distinguished name or if
-     *                                  secrets is empty.
+     * @throws IllegalArgumentException if the given string is not a valid X.500 distinguished name or if secrets is
```

Comment:
we do only throw this if secrets is empty, right?

Reply:
Added validation for empty authId too in a new commit.

Reply:
My point is that we actually accept an auth-id that is not a valid X.500 distinguishedName.
Instead, we reject an empty auth-id ...